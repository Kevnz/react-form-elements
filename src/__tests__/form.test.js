import React from 'react'
import { render, fireEvent, cleanup } from 'react-testing-library'

import {
  TextBox,
  Checkbox,
  Range,
  DateTime,
  Telephone,
  DropDown,
  Option,
  OptionGroup,
  Radio,
  Form,
} from '../index'

const App = ({ onSubmit }) => {
  return (
    <Form
      name="testForm"
      onSubmit={e => {
        onSubmit(e)
      }}
    >
      <TextBox label="My Label" name="myTextBox" />
      <Checkbox
        label="My Checkbox"
        initialValue="The True"
        checked={false}
        name="myCheckBox"
      />
      <Radio label="My Radio" name="myRadio" checked={false} />
      <DateTime label="My Date" name="myDate" />
      <Range label="My Range" name="myRange" />
      <Telephone label="Telephone" name="myTelephone" />
      <DateTime label="My DateTime" type="datetime" name="myDateTime" />
      <DateTime label="My Month" type="month" name="myMonth" />
      <DateTime label="My Week" type="week" name="myWeek" />
      <DateTime
        label="My DateTime"
        type="datetime-local"
        name="myDateTimeLocal"
      />
      <DateTime label="My Time" type="time" name="myTime" />
      <DropDown
        label="My Drop Down"
        initialValue="2"
        data-testid="dd1"
        name="myDropDown"
      >
        <OptionGroup label="First Group">
          <Option initialValue="1">First</Option>
          <Option initialValue="2">Second</Option>
          <Option initialValue="3" label="Third" />
        </OptionGroup>
        <OptionGroup label="Second Group">
          <Option initialValue="11">Second First</Option>
          <Option initialValue="12">Second Second</Option>
          <Option initialValue="13" label="Second Third" />
        </OptionGroup>
        <OptionGroup label="Third Group">
          <Option initialValue="21">Third First</Option>
          <Option initialValue="22">Third Second</Option>
          <Option initialValue="23" label="Third Third" />
        </OptionGroup>
      </DropDown>
      <button>Save</button>
    </Form>
  )
}

describe('The form components as a form', () => {
  afterEach(cleanup)
  it('should render and change accordingly', () => {
    const CHANGED_TEXTBOX = 'Changed Value'
    const { container, getByLabelText, getByTestId, getByText } = render(
      <App
        onSubmit={e => {
          expect(e.myTextBox).toBe(CHANGED_TEXTBOX)
          expect(e.myDropDown).toBe('22')
        }}
      />
    )
    const textBox = getByLabelText('My Label')
    const dd = getByTestId('dd1')
    const button = getByText('Save')
    expect(container.firstChild).toMatchSnapshot()
    const dropDownValue = dd.value
    expect(dropDownValue).toBe('2')
    fireEvent.change(textBox, { target: { value: CHANGED_TEXTBOX } })
    fireEvent.change(dd, { target: { value: '22' } })
    fireEvent.click(button)
  })
})
