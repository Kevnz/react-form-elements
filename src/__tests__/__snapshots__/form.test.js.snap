// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The form components as a form should render and change accordingly 1`] = `
<form
  name="testForm"
>
  <div
    class="form-row_container"
  >
    <label>
      My Label
      <input
        name="myTextBox"
        type="text"
        value=""
      />
    </label>
  </div>
  <div
    class="form-row_container"
  >
    <label>
      My Checkbox
      <input
        name="myCheckBox"
        type="checkbox"
      />
    </label>
  </div>
  <div
    class="form-row_container"
  >
    <label>
      My Radio
      <input
        name="myRadio"
        type="radio"
      />
    </label>
  </div>
  <div
    class="form-row_container"
  >
    <label>
      My Date
      <input
        name="myDate"
        type="date"
        value=""
      />
    </label>
  </div>
  <div
    class="form-row_container"
  >
    <label>
      My Range
      <input
        max="100"
        min="0"
        name="myRange"
        step="1"
        type="range"
        value="50"
      />
    </label>
    <span>
      50
    </span>
  </div>
  <div
    class="form-row_container"
  >
    <label>
      Telephone
      <input
        name="myTelephone"
        type="tel"
        value=""
      />
    </label>
  </div>
  <div
    class="form-row_container"
  >
    <label>
      My DateTime
      <input
        name="myDateTime"
        type="datetime"
        value=""
      />
    </label>
  </div>
  <div
    class="form-row_container"
  >
    <label>
      My Month
      <input
        name="myMonth"
        type="month"
        value=""
      />
    </label>
  </div>
  <div
    class="form-row_container"
  >
    <label>
      My Week
      <input
        name="myWeek"
        type="week"
        value=""
      />
    </label>
  </div>
  <div
    class="form-row_container"
  >
    <label>
      My DateTime
      <input
        name="myDateTimeLocal"
        type="datetime-local"
        value=""
      />
    </label>
  </div>
  <div
    class="form-row_container"
  >
    <label>
      My Time
      <input
        name="myTime"
        type="time"
        value=""
      />
    </label>
  </div>
  <div
    class="form-row_container"
  >
    <label>
      My Drop Down
      <select
        data-testid="dd1"
        name="myDropDown"
      >
        <optgroup
          label="First Group"
        >
          <option
            value="1"
          >
            First
          </option>
          <option
            value="2"
          >
            Second
          </option>
          <option
            value="3"
          >
            Third
          </option>
        </optgroup>
        <optgroup
          label="Second Group"
        >
          <option
            value="11"
          >
            Second First
          </option>
          <option
            value="12"
          >
            Second Second
          </option>
          <option
            value="13"
          >
            Second Third
          </option>
        </optgroup>
        <optgroup
          label="Third Group"
        >
          <option
            value="21"
          >
            Third First
          </option>
          <option
            value="22"
          >
            Third Second
          </option>
          <option
            value="23"
          >
            Third Third
          </option>
        </optgroup>
      </select>
    </label>
  </div>
  <button>
    Save
  </button>
</form>
`;
